src/signals.jl

```diff
@@ -32,279 +34,102 @@ function onda_sample_type_from_julia_type(T::Type)
+    return x
 end
 
-#####
-##### `SamplesInfo`
-#####
-
-"""
-    SamplesInfo(; kind, channels, sample_unit,
-                sample_resolution_in_unit, sample_offset_in_unit,
-                sample_type, sample_rate,
-                validate::Bool=Onda.validate_on_construction())
-    SamplesInfo(kind, channels, sample_unit,
-                sample_resolution_in_unit, sample_offset_in_unit,
-                sample_type, sample_rate;
-                validate::Bool=Onda.validate_on_construction())
-    SamplesInfo(signals_table_row; validate::Bool=Onda.validate_on_construction())
-
-Return a `SamplesInfo` instance whose fields are a subset of a `*.onda.signals.arrow` row:
-
-- `kind`
-- `channels`
-- `sample_unit`
-- `sample_resolution_in_unit`
-- `sample_offset_in_unit`
-- `sample_type`
-- `sample_rate`
-
-The `SamplesInfo` struct bundles together the fields of a `*.onda.signals.arrow` row that are
-intrinsic to a signal's sample data, leaving out extrinsic file or recording information.
-This is useful when the latter information is irrelevant or does not yet exist (e.g. if
-sample data is being constructed/manipulated in-memory without yet having been serialized).
-
-Bundling these fields together under a common type facilitates dispatch for various Onda API
-functions. Additionally:
-
-- If `validate` is `true`, then `Onda.validate` is called on new instances upon construction.
-
-- The provided `sample_type` may be either an Onda-compliant string or a `DataType`. If it is
-  a string, it will be converted to its corresponding `DataType`.
-"""
-struct SamplesInfo{K<:AbstractString,
-                   C<:AbstractVector{<:AbstractString},
-                   U<:AbstractString,
-                   R<:LPCM_SAMPLE_TYPE_UNION,
-                   O<:LPCM_SAMPLE_TYPE_UNION,
-                   S<:LPCM_SAMPLE_TYPE_UNION,
-                   SR<:LPCM_SAMPLE_TYPE_UNION}
-    kind::K
-    channels::C
-    sample_unit::U
-    sample_resolution_in_unit::R
-    sample_offset_in_unit::O
-    sample_type::Type{S}
-    sample_rate::SR
-    function SamplesInfo(kind::K, channels::C, sample_unit::U,
-                         sample_resolution_in_unit::R,
-                         sample_offset_in_unit::O,
-                         sample_type, sample_rate::SR;
-                         validate::Bool=Onda.validate_on_construction()) where {K,C,U,R,O,SR}
-        S = sample_type isa Type ? sample_type : julia_type_from_onda_sample_type(sample_type)
-        info = new{K,C,U,R,O,S,SR}(kind, channels, sample_unit, sample_resolution_in_unit,
-                                   sample_offset_in_unit, S, sample_rate)
-        validate && Onda.validate(info)
-        return info
+function _validate_signal_channels(x)
+    for c in x
+        is_lower_snake_case_alphanumeric(c, ('-', '.')) || throw(ArgumentError("invalid channel name (must be lowercase/snakecase/alphanumeric): $c"))
     end
+    return x
 end
```

Comment:
Use broadcasting where `_validate_signal_channels` was used:
```suggestion
function _validate_signal_channel(x)
    is_lower_snake_case_alphanumeric(x, ('-', '.')) || throw(ArgumentError("invalid channel name (must be lowercase/snakecase/alphanumeric): $x"))
    return x
end
```

Reply:
0d1539795e45fe1b6d9195a639fcc492a9c31e64